csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -219,6 +278,10 @@ module DataFlowPrivateCached {
     exists(ControlFlow::Nodes::ElementNode cfn | node = TImplicitDelegateOutNode(cfn, _) |
       result = cfn.getEnclosingCallable()
     )
+    or
+    exists(ControlFlow::Nodes::ElementNode cfn | node = TMallocNode(cfn) |
```

Comment:
These types really look like they should be split up into classes. I appreciate this is a caching-in-same-stage hack but can we put a little class hierarchy in here?

Reply:
OK, I have changed these to be defined by dispatch instead. Cache behaviour should be the same.